utils/src/test/scala/za/co/absa/enceladus/utils/types/DefaultsByFormatSuite.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.utils.types
+
+import org.scalatest.funsuite.AnyFunSuite
+
+class DefaultsByFormatSuite extends AnyFunSuite {
+  test("Format specific timestamp time zone override exists") {
+    val default = new DefaultsByFormat("testFormat")
+    assert(default.getDefaultTimestampTimeZone.contains("Africa/Johannesburg"))
+  }
+
+  test("Format specific timestamp time zone override does not exists") {
+    val default = new DefaultsByFormat("testFormatNotExisting")
+    assert(default.getDefaultTimestampTimeZone.contains("CET"))
+  }
+
+  test("Format specific timestamp time zone override is not a valid time zone id") {
+    intercept[IllegalStateException] {
+      new DefaultsByFormat("testFormatFail")
+    }
+  }
+
+  test("Format specific date time zone override does not exists") {
+    val default = new DefaultsByFormat("testFormat")
+    assert(default.getDefaultDateTimeZone.isEmpty)
```

Comment:
The tests make sense, but unfortunately, I find use-cases of _empty_ `default.getDefaultTimestampTimeZone` and _non-empty_ `default.getDefaultDateTimeZone` missing.

That, I think could be done by passing a custom config that differs from the by-default-loaded by e.g.
```scala
val defaults2 = new DefaultsByFormat("xxx") {
   override protected val config: ConfigReader = new ConfigReader(
     ConfigFactory.empty().withValue("defaultDateTimeZone", ConfigValueFactory.fromAnyRef("PST"))
       // or .withValue("defaultDateTimeZone-myAwesomeFormat", ConfigValueFactory.fromAnyRef("JST"))
   )
 }
```

Reply:
Ha good hint. I didn't know this way of slipping in configuration settings. I will try to apply it... 👍

Reply:
It seems I can't do this. The `config` is access during object construction, so if overridden, it's not created yet, therefore getting null-pointer exception.

Reply:
Aha, sorry. I see you found a nice solution, though.